addon/model.js

```diff
@@ -167,21 +163,45 @@ function resolveValue(key, value, modelName, store, schema, model, parentIdx) {
 
 // ie an array of nested models
 function resolvePlainArray(key, value, modelName, store, schema, model) {
-  if (value == null) {
-    return new Array(0);
+  let resolvedArray = new Array(0);
+  if (value && value.length > 0) {
+    resolvedArray = value.map((value, idx) =>
+      resolveValue(key, value, modelName, store, schema, model, idx)
+    );
   }
 
-  return value.map((value, idx) => resolveValue(key, value, modelName, store, schema, model, idx));
+  return M3PlainArray.create({
+    content: A(resolvedArray),
+    _replace(idx, removeAmt, newItems) {
```

Comment:
I'm not a fan of passing in `_replace` here; I think it obscures the way the arrays function too much.

`resolve` can be imported by the various record arrays: i'd rather we create the arrays with the relevant context that they stash during `init` than pass in a function that has that context in a closure.